modules/apps/application-list/application-list-user-personal-site-permissions/src/main/java/com/liferay/application/list/user/personal/site/permissions/internal/model/listener/CompanyModelListener.java

```diff
@@ -67,6 +108,219 @@ public void onAfterCreate(Company company) {
 	private PortletLocalService _portletLocalService;
 
 	@Reference
+	private ResourcePermissionLocalService _resourcePermissionLocalService;
+
+	@Reference
+	private RoleLocalService _roleLocalService;
+
+	private ServiceTrackerMap<String, PanelApp> _serviceTrackerMap;
 	private UserPersonalSitePermissions _userPersonalSitePermissions;
 
+	private class PanelAppServiceTrackerCustomizer
+		implements EagerServiceTrackerCustomizer<PanelApp, PanelApp> {
+
+		public PanelAppServiceTrackerCustomizer(BundleContext bundleContext) {
+			_bundleContext = bundleContext;
+		}
+
+		@Override
+		public PanelApp addingService(
+			ServiceReference<PanelApp> serviceReference) {
+
+			PanelApp panelApp = _bundleContext.getService(serviceReference);
+
+			try {
+				Portlet portlet = panelApp.getPortlet();
+
+				if (portlet == null) {
+					portlet = _portletLocalService.getPortletById(
+						panelApp.getPortletId());
+				}
+
+				if (portlet == null) {
+					Class<?> panelAppClass = panelApp.getClass();
+
+					_log.error(
+						StringBundler.concat(
+							"Unable to get portlet ", panelApp.getPortletId(),
+							" for panel app ", panelAppClass.getName()));
+
+					return panelApp;
+				}
+
+				_userPersonalSitePermissions.initPermissions(portlet);
+
+				return panelApp;
+			}
+			catch (Throwable throwable) {
+				_bundleContext.ungetService(serviceReference);
+
+				throw throwable;
+			}
+		}
+
+		@Override
+		public void modifiedService(
+			ServiceReference<PanelApp> serviceReference, PanelApp panelApp) {
+		}
+
+		@Override
+		public void removedService(
+			ServiceReference<PanelApp> serviceReference, PanelApp panelApp) {
+
+			_bundleContext.ungetService(serviceReference);
+		}
+
+		private final BundleContext _bundleContext;
+
+	}
+
+	private class UserPersonalSitePermissions {
```

Comment:
you do not need this inner class... just merge the `UserPersonalSitePermissions` methods into the `CompanyModelListener` sounds good.